boot/bootutil/src/bootutil_misc.c

```diff
@@ -837,3 +837,30 @@ boot_set_confirmed(void)
     flash_area_close(fap);
     return rc;
 }
+
+/**
+ * Populates the version information of the
+ * currently installed primary application
+ *
+ * @param[in] version Destination version structure buffer
+ * @return 0 on success; nonzero on failure.
+ */
+int boot_get_current_version(struct image_version *version)
+{
+    struct boot_loader_state state;
+    struct boot_status status;
+    struct image_header hdr;
+    int rc;
+
+    assert(version != NULL);
```

Comment:
This is not OK. You should fail when someone passes `version` == `NULL`, not cause a fault. Also `assert` could be opted out by defining `NDEBUG` which would make this a no-opt. Building with `NDEBUG` is currently unsupported by MCUboot, but the issue was already raised in the Mynewt community, so until we fix it, at least don't add new code which depends on `assert`.

Reply:
I have removed the call to `assert` and instead now return `BOOT_EBADARGS` if `NULL` is passed in.